lib/cmds/template_generator.js

```diff
@@ -6,22 +6,38 @@ class TemplateGenerator {
     this.templateName = templateName;
   }
 
+  downloadAndGenerate(uri, destinationFolder, name) {
+    const self = this;
+    let {url, filePath} = this.getExternalProject(uri);
+    let tmpFilePath = fs.tmpDir(filePath);
+    console.log(__('Installing Template from ' + uri + '....').green);
+
+    fs.mkdirpSync(utils.dirname(tmpFilePath));
+    utils.downloadFile(url, tmpFilePath, () => {
+      let fspath = utils.joinPath(destinationFolder, name);
+      const decompress = require('decompress');
```

Comment:
You could use utils.extractZip that you added below

Reply:
ups, I did that for debuggin and forgot to revert back to utils